CMakeLists.txt

```diff
@@ -331,6 +305,80 @@ add_swift_library(Foundation
                     CoreFoundation
                     $<$<PLATFORM_ID:Windows>:CoreFoundationResources>)
 
+add_swift_library(FoundationNetworking
+                  MODULE_NAME
+                    FoundationNetworking
+                  MODULE_LINK_NAME
+                    FoundationNetworking
+                  MODULE_PATH
+                    ${CMAKE_CURRENT_BINARY_DIR}/swift/FoundationNetworking.swiftmodule
+                  SOURCES
+                  Foundation/Boxing.swift
+                  Foundation/NSURLRequest.swift
+                  Foundation/HTTPCookie.swift
+                  Foundation/HTTPCookieStorage.swift
+                  Foundation/URLAuthenticationChallenge.swift
+                  Foundation/URLCache.swift
+                  Foundation/URLCredential.swift
+                  Foundation/URLCredentialStorage.swift
+                  Foundation/URLProtectionSpace.swift
+                  Foundation/URLProtocol.swift
+                  Foundation/URLRequest.swift
+                  Foundation/URLResponse.swift
+                  Foundation/URLSession/BodySource.swift
+                  Foundation/URLSession/Configuration.swift
+                  Foundation/URLSession/http/HTTPMessage.swift
+                  Foundation/URLSession/http/HTTPURLProtocol.swift
+                  Foundation/URLSession/libcurl/EasyHandle.swift
+                  Foundation/URLSession/libcurl/libcurlHelpers.swift
+                  Foundation/URLSession/libcurl/MultiHandle.swift
+                  Foundation/URLSession/Message.swift
+                  Foundation/URLSession/NativeProtocol.swift
+                  Foundation/URLSession/NetworkingSpecific.swift
+                  Foundation/URLSession/ftp/FTPURLProtocol.swift
+                  Foundation/URLSession/TaskRegistry.swift
+                  Foundation/URLSession/TransferState.swift
+                  Foundation/URLSession/URLSession.swift
+                  Foundation/URLSession/URLSessionConfiguration.swift
+                  Foundation/URLSession/URLSessionDelegate.swift
+                  Foundation/URLSession/URLSessionTask.swift
+                  TARGET
+                    ${CMAKE_C_COMPILER_TARGET}
+                  CFLAGS
+                    ${MSVCRT_C_FLAGS}
+                    ${deployment_target}
+                    ${deployment_enable_libdispatch}
```

Comment:
I believe that I have cleaned this up as well.  This was previously needed as there was a mode for `CoreFoundation` to be built without libdispatch which is no longer supported.